### PR TITLE
KG contribution to Harris functional forces

### PR DESCRIPTION
**Pull request by fbelle**
Adding Kim-Gordon subsystem DFT support (KG_METHOD ATOM and EMBEDDING) for Harris functional energy correction force calculations

*Prettify

### CODE REVIEW

**Review by dev-zero — CHANGES_REQUESTED** [review]
other than commented lines LGTM

File: src/kg_correction.F [hunk 1]
```diff
@@ -597,4 +692,122 @@ SUBROUTINE kg_ekin_atomic(qs_env, ks_matrix, ekin_mol)
...
+! **************************************************************************************************
+!> \brief   Calculation of second derivative xc-potential
+!> \param   qs_env ...
+!> \param   vxc [OUT], contains partially integrated second derivative
+!>          taken with respect to rho, evluated in rho and folded with rho1
+!> \param rho ...
+!> \param rho1 ...
+!> \param xc_section ...
+!> \date    11.2019
+!> \param
```
Comment (dev-zero) [inline src/kg_correction.F]: interesting that `doxify` doesn't remove those lines

File: src/kg_correction.F [hunk 2]
```diff
@@ -597,4 +692,122 @@ SUBROUTINE kg_ekin_atomic(qs_env, ks_matrix, ekin_mol)
...
+      TYPE(pw_p_type), DIMENSION(:), POINTER             :: rho1_g, rho1_r, rho_r, tau_pw, vxc_rho
+      TYPE(pw_pool_type), POINTER                        :: auxbas_pw_pool
+      TYPE(section_vals_type), POINTER                   :: xc_fun_section
+      TYPE(xc_derivative_set_type), POINTER              :: deriv_set
+      TYPE(xc_rho_cflags_type)                           :: needs
+      TYPE(xc_rho_set_type), POINTER                     :: rho1_set, rho_set
+
+      CALL timeset(routineN, handle)
+      NULLIFY (logger)
+      logger => cp_get_default_logger()
```
Comment (dev-zero) [inline src/kg_correction.F]: not used, or did I miss it?

File: src/kg_correction.F [hunk 3]
```diff
@@ -597,4 +692,122 @@ SUBROUTINE kg_ekin_atomic(qs_env, ks_matrix, ekin_mol)
...
+!> \param rho1 ...
+!> \param xc_section ...
+!> \date    11.2019
+!> \param
+!> \param
+!> \author  fbelle
+! **************************************************************************************************
+   SUBROUTINE calculate_kernel(qs_env, vxc, rho, rho1, xc_section)
+
+      TYPE(qs_environment_type), POINTER                 :: qs_env
```
[the remark this answers is not in the document]
Comment (fbelle) [reply src/kg_correction.F:707]: Renamed to create_kernel to avoid confusion. Added description about allocation of vxc.

**Comment by dev-zero** [comment]
you still might want to add a regtest for this at some point

**Comment by fbelle** [comment]
Thank you! Adding a regtest is indeed on my to do list!

**Pull request merged**